server/verify.go

```diff
@@ -183,6 +152,45 @@ func VerifyAttestation(attestation *pb.Attestation, opts VerifyOpts) (*pb.Machin
 	return nil, fmt.Errorf("attestation does not contain a supported quote")
 }
 
+// ParseAndValidateAK parses and validate AK cert in the attestation, and populate GCE instance info.
```

Comment:
Ideally, we have composable funcs where we can verify the attestation key against the Attestation proto and VerifyOpts and then extract info from it. So I suggest a bit more refactoring:
1. call this `ValidateAK`
2. Have `ValidateAK` just call `validateAKPub` and `validateAKCert`
3. refactor `validateAKPub` and `validateAKCert` to only return error
4. refactor `validateAKCert` to not call `getInstanceInfoFromExtensions`
5. in `VerifyAttestation` call `ValidateAK` and then `getInstanceInfoFromExtensions` (or `GetGCEInstanceInfo` if cert != nil)

Reply:
Done